kratos/python_scripts/line_output_process.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# other imports
+from multiple_points_output_process import MultiplePointsOutputProcess
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return LineOutputProcess(Model, settings["Parameters"])
+
+class LineOutputProcess(KratosMultiphysics.Process):
+    """This process writes output for several points along a line to a file
+    Internally it holds an object of type "MultiplePointsOutputProcess"
+    Usage:
+        - specifying start and end point defining the line
+        - specifying the number of sampling points alond the line (start and end points will be included)
+    """
+    def __init__(self, model, params):
+
+        default_settings = KratosMultiphysics.Parameters('''{
+            "start_point"               : [],
+            "end_point"                 : [],
+            "number_of_sampling_points" : 3,
+            "model_part_name"           : "",
+            "output_file_name"          : "",
+            "output_variables"          : [],
+            "entity_type"               : "element"
+        }''')
+
+        params.ValidateAndAssignDefaults(default_settings)
+
+        # retrieving the positions defining the line entity as well as the number of sampling points
+        start_point_position = params["start_point"].GetVector()
+        if start_point_position.Size() != 3:
+            raise Exception('The start point position has to be provided with 3 coordinates!')
+        end_point_position = params["end_point"].GetVector()
+        if end_point_position.Size() != 3:
+            raise Exception('The end point position has to be provided with 3 coordinates!')
+        number_of_sampling_points = params["number_of_sampling_points"].GetInt()
+        if number_of_sampling_points < 1:
```

Comment:
Here number_of_sampling_points should be at least 2. Otherwise you get div0 below.